src/transformState.js

```diff
@@ -5,7 +5,36 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData } = action;
+    let { keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties': {
+        for (const [key, value] of Object.entries(extraData)) {
+          state[key] = value;
+        }
```

Comment:
Objects already have method that is designed to `assign` properties from other objects)